### PR TITLE
Fix #1899: Thoroughly test ItemsSelectionInputEqualsRuleClassifier

### LINKED ISSUE #1899: Thoroughly test ItemsSelectionInputEqualsRuleClassifier
Since the ItemsSelectionInputEqualsRuleClassifier was copied from Oppia web, it didn't have corresponding tests added to verify correctness. These should be ported over from Oppia web as well to ensure that each classifier has working functionality.

Useful reference - https://github.com/oppia/oppia-android/tree/develop/domain/src/test/java/org/oppia/android/domain/classify/rules

### PR DESCRIPTION
**Pull request by peculiaruc**
## Explanation
Fix #1899, Format ItemsSelectionInputEqualsRuleClassifierProviderTest

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by peculiaruc** [comment]
Screenshot from 2020-11-13 18-33-26

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    createSetOfHtmlString(
+      InteractionObjectTestBuilder.createHtmlStringList("item ab")
+    )
+
+  private val TEST_HTML_STRING_SET_UPPERCASE =
+    createSetOfHtmlString(
+      InteractionObjectTestBuilder.createHtmlStringList("item AB")
+    )
+
+  private val TEST_HTML_STRING_SET_MIXEDLOWERCASE =
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Nit:
`TEST_HTML_STRING_SET_MIXED_LOWERCASE `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    createSetOfHtmlString(
+      InteractionObjectTestBuilder.createHtmlStringList("item aA ")
+    )
+  private val TEST_HTML_STRING_SET_MIXEDUPPERCASE =
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Nit:
`TEST_HTML_STRING_SET_MIXED_LOWERCASE `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    createSetOfHtmlString(
+      InteractionObjectTestBuilder.createHtmlStringList("item bB")
+    )
+
+  private val TEST_NON_NEGATIVE_VALUE_3 =
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: This name looks different than whats is set in the value. Can you please recheck this?

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    itemSelectionInputEqualsRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testLowercaseStringAnswer_samevalue_bothValuesMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Nit:
`testLowercaseStringAnswer_sameValue_bothValuesMatch`

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_LOWERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_lowercaseAnswer_withlowercaseAnswerInput_bothValuesMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_lowercaseAnswer_withLowercaseAnswerInput_bothValuesMatch`

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_LOWERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_lowercaseAnswer_withuppercaseAnswerInput_bothValuesNotMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_lowercaseAnswer_withUppercaseAnswerInput_bothValuesNotMatch `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_LOWERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_uppercaseAnswer_withlowercaseAnswerInput_bothValuesNotMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_uppercaseAnswer_withLowercaseAnswerInput_bothValuesNotMatch `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_LOWERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_uppercaseAnswer_withuppercaseAnswerInput_bothValuesMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_uppercaseAnswer_withUppercaseAnswerInput_bothValuesMatch `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_UPPERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_mixedlowercaseAnswer_withmixeduppercaseAnswerInput_bothValuesNotMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_mixedlowercaseAnswer_withMixedUppercaseAnswerInput_bothValuesNotMatch `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_MIXEDUPPERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_nonNegativeAnswer_withuppercaseAnswerInput_bothValuesMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_nonNegativeAnswer_withUppercaseAnswerInput_bothValuesMatch `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_UPPERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_nonNegativeAnswer_withlowercaseAnswerInput_bothValuesNotMatch() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testEquals_nonNegativeAnswer_withLowercaseAnswerInput_bothValuesNotMatch`

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+      inputEqualsRuleClassifierProvider.matches(
+        answer = TEST_HTML_STRING_SET_LOWERCASE,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testlowercaseAnswer_missingInput_throwsException() {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: `testLowercaseAnswer_missingInput_throwsException `

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+        inputs = inputs
+      )
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'x'")
+  }
+
+  private fun createSetOfHtmlString(value: StringList): InteractionObject {
```
Comment (rt4914) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: This should be created in `InteractionObjectTestBuilder`

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL @peculiaruc

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 13 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: @rt4914, It is not there. The file is not assigned to me. I know for me to add any code in any file, that file has to be assigned to me. please what should be my next action? Thanks.

Comment (rt4914) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: @peculiaruc Can you access this file and make changes? https://github.com/oppia/oppia-android/blob/develop/domain/src/test/java/org/oppia/android/domain/classify/InteractionObjectTestBuilder.kt

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 2 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 3 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 4 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 5 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 6 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 7 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 8 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 9 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 10 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 11 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 12 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 13 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

File: domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt (hunk 1 above)
Comment (peculiaruc) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/itemselectioninput/ItemSelectionInputEqualsRuleClassifierProviderTest.kt]: Done

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @peculiaruc

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by rt4914** [comment]
@peculiaruc Please resolve merge conflicts so that we can merge this. Thanks.

File: .idea/gradle.xml [hunk 14]
```diff
@@ -20,7 +20,8 @@
           </set>
         </option>
         <option name="resolveModulePerSourceSet" value="false" />
+        <option name="useQualifiedModuleNames" value="true" />
       </GradleProjectSettings>
     </option>
   </component>
-</project>
+</project>
```
Comment (rt4914) [inline .idea/gradle.xml]: Please revert changes from this file.

File: .idea/misc.xml [hunk 15]
```diff
@@ -1,9 +1,9 @@
...
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (rt4914) [inline .idea/misc.xml]: Please revert changes from this file.

File: build.gradle [hunk 16]
```diff
@@ -9,7 +9,7 @@ buildscript {
     gradlePluginPortal()
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:3.6.1'
+    classpath 'com.android.tools.build:gradle:4.1.1'
```
Comment (rt4914) [inline build.gradle]: Please revert changes from this file.

File: gradle/wrapper/gradle-wrapper.properties [hunk 17]
```diff
@@ -1,6 +1,6 @@
-#Sun Mar 29 11:36:23 IST 2020
+#Fri Nov 13 19:44:58 WAT 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
```
Comment (rt4914) [inline gradle/wrapper/gradle-wrapper.properties]: Please revert changes from this file.

**Review by rt4914 — CHANGES_REQUESTED** [review]
With latest merge there are some extra file changes, please revert them. Thanks @peculiaruc

File: .idea/gradle.xml (hunk 14 above)
Comment (peculiaruc) [reply .idea/gradle.xml]: Done

File: .idea/misc.xml (hunk 15 above)
Comment (peculiaruc) [reply .idea/misc.xml]: Done

File: build.gradle (hunk 16 above)
Comment (peculiaruc) [reply build.gradle]: Done

File: gradle/wrapper/gradle-wrapper.properties (hunk 17 above)
Comment (peculiaruc) [reply gradle/wrapper/gradle-wrapper.properties]: Done

**Comment by rt4914** [comment]
@peculiaruc This PR is showing 63 files changed.
You will need to update your PR branch and your forked-`develop` so that it shows only those files which you have changed. Thanks.

**Comment by peculiaruc** [comment]
@rt4914 I made changes and pushed here but I can't find the codes

**Comment by peculiaruc** [comment]
> @rt4914 I made changes and pushed here but I can't find the codes

NeverMind I have seen it

**Pull request closed without merging**